File: java/IEC61970/Base/Wires/ProtectedSwitch.java


import TC57CIM.IEC61970.Base.Domain.CurrentFlow;
import TC57CIM.IEC61970.Base.Protection.RecloseSequence;

/**
 * A ProtectedSwitch is a switching device that can be operated by
 * ProtectionEquipment.
 * @author pmora
 * @updated 15-Dec-2023 1:39:42 PM
 */
public class ProtectedSwitch extends Switch {

	/**
	 * The maximum fault current a breaking device can break safely under prescribed
	 * conditions of use.
	 */
	private CurrentFlow breakingCapacity;

	public ProtectedSwitch(){

	}

	public void finalize() throws Throwable {
		super.finalize();
	}

	public CurrentFlow getbreakingCapacity(){
		return breakingCapacity;
	}

	public RecloseSequence getRecloseSequences(){
		return RecloseSequences;
	}

	/**
	 * 
	 * @param newVal
	 */
	public void setbreakingCapacity(CurrentFlow newVal){
		breakingCapacity = newVal;
	}

	/**
	 * 
	 * @param newVal
	 */
	public void setRecloseSequences(RecloseSequence newVal){
		RecloseSequences = newVal;
	}

}

File: py/IEC61970/Base/Wires/ProtectedSwitch.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Domain.CurrentFlow import CurrentFlow
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Wires.Switch import Switch


class RecloseSequence:
    pass


class ProtectedSwitch(Switch):

    def __init__(self) -> None:
        super().__init__()
        self.breaking_capacity = CurrentFlow()
        self.reclose_sequences = RecloseSequence()

    def get_breaking_capacity(self) -> CurrentFlow:
        return self.breaking_capacity

    def get_reclose_sequences(self) -> RecloseSequence:
        return self.reclose_sequences

    def set_breaking_capacity(self, new_val: CurrentFlow) -> None:
        self.breaking_capacity = new_val

    def set_reclose_sequences(self, new_val: RecloseSequence) -> None:
        self.reclose_sequences = new_val
